DokuwikiXMLExport.php

```diff
@@ -0,0 +1,222 @@
+     *
+     * @param $start integer Determines the first item (offset) to be exported.
+     * @param $count integer Determines the interval size / number of items to be exported.
+     * @return string Returns the XML.
+     */
+    public function generateXMLExport($start, $count)
+    {
+        $indexer = new Doku_Indexer();
+        $pagesAndDeletedPages = $indexer->getPages();
+
+        $excludedPages = $this->splitConfigToArray($conf['plugin']['findologicxmlexport']['excludePages']);
+        $pages = null;
+        foreach ($pagesAndDeletedPages as $page) {
+            if (p_get_metadata($page)['description']) { // Only get pages with content
+                if (!in_array($page, $excludedPages)) { // Exclude pages from config
+                    $pages[] = $page;
+                }
+            }
+        }
+        $exporter = Exporter::create(Exporter::TYPE_XML, $count);
+
+        $total = count($pages);
+
+        // The count can't be higher then the total number of pages.
+        $count = min($total, $count);
+
+
+        // If count + start is higher then total, then something went totally wrong.
+        if (($count + $start) > $total) {
+            throw new \InvalidArgumentException("Call method " . __METHOD__ . " in class " . __CLASS__ . " failed while trying to validate start and count values");
+        }
+
+        $items = array();
+        for ($i = $start; $i < $start + $count; $i++) {
+            $item = $exporter->createItem($i);
+
+            $name = new Name();
+            $name->setValue($this->getName($pages, $i));
+            $item->setName($name);
+
+            $summary = new Summary();
+            $summary->setValue($this->getSummaryAndDescription($pages, $i));
+            $item->setSummary($summary);
+
+            $description = new Description();
+            $description->setValue($this->getSummaryAndDescription($pages, $i));
+            $item->setDescription($description);
+
+            $price = new Price();
+            $price->setValue(self::PRICE_PLACEHOLDER);
+            $item->setPrice($price);
+
+            $Url = new Url();
+            $Url->setValue($this->getUrl($pages, $i));
+            $item->setUrl($Url);
+
+            $dateAdded = new DateAdded();
+            $dateAdded->setDateValue($this->getDateAdded($pages, $i));
+            $item->setDateAdded($dateAdded);
+
+            $item->addOrdernumber(new Ordernumber($this->getOrdernumber($pages, $i)));
+            $items[] = $item;
+
+            $attributeCategory = new Attribute(self::CATEGORY_KEY, $this->getAttributesCategory($pages, $i));
+            $item->addAttribute($attributeCategory);
+        }
+        return $exporter->serializeItems($items, $start, $total);
+    }
+
+    protected $conf;
+
+    public function __construct($conf)
+    {
+        $this->conf = $conf;
+    }
+
+    /**
+     * Gets the ID of the DokuWiki page and sets the ordernumber equal to it.
```

Comment:
> sets the ordernumber equal to it.

Strictly speaking, that's not what the method does. It just returns the ID / ordernumber. The caller of the method sets it in the export.